libmscore/cmd.cpp

```diff
@@ -2947,6 +2947,209 @@ void Score::cmdSlashRhythm()
+                  int offset = 0;
+                  Interval interval = h->staff()->part()->instrument(h->tick())->transpose();
+                  if (!concertPitch)
+                        offset = interval.chromatic;
+                  notes = r.generateNotes(h->rootTpc(), h->baseTpc(),
+                        literal, voicing, offset);
+                  }
+            QMapIterator<int, int> i(notes); //add notes to chord
+            while (i.hasNext()) {
+                  i.next();
+                  Note* note = new Note(this);
+                  NoteVal nval;
+                  nval.pitch = i.key();
+                  if (concertPitch)
+                        nval.tpc1 = i.value();
+                  else
+                        nval.tpc2 = i.value();
+                  chord->add(note); //add note first to set track and such
+                  note->setNval(nval, tick);
+                  }
+
+            setChord(seg, h->track(), chord, duration); //add chord using template
+            delete chord;
+            }
+      }
+
+//---------------------------------------------------------
+//   setChord
+//    return segment of last created chord
+//---------------------------------------------------------
+Segment* Score::setChord(Segment* segment, int track, Chord* chordTemplate, Fraction dur, Direction stemDirection)
+      {
+      Q_ASSERT(segment->segmentType() == SegmentType::ChordRest);
+
+      Fraction tick = segment->tick();
+      Chord* nr     = 0; //current added chord used so we can select the last added chord and so we can apply ties
+      std::vector<Tie*> tie(chordTemplate->notes().size()); //keep pointer to a tie for each note in the chord in case we need to tie notes
+      ChordRest* cr = toChordRest(segment->element(track)); //chord rest under the segment for the specified track
+
+      bool addTie = false;
+
+      Measure* measure = 0;
+      //keep creating chords and tieing them until we created the full duration asked for (dur)
+      for (;;) {
+            if (track % VOICES)
+                  expandVoice(segment, track);
+
+            Tuplet* t = cr ? cr->tuplet() : 0;
+            Fraction tDur = segment->ticks();
+            Segment* seg = segment->next();
+
+            //we need to get a correct subduration so that makeGap can function properly
+            //since makeGap() takes "normal" duration rather than actual length
+            while (seg) {
+                  if (seg->segmentType() == SegmentType::ChordRest) {
+                        //design choice made to keep multiple notes across a tuplet as tied single notes rather than combining them
+                        //since it's arguably more readable, but the other code is still here (commented)
+                        ChordRest* testCr = toChordRest(seg->element(track));
+                        /*if (!!t ^ (testCr && testCr->tuplet())) //stop if we started with a tuplet and reach something that's not a tuplet,
```

Comment:
Stylistically, I prefer // for each line, if #if 0, for code we want to leave in place but commented out for whatever reason.  As it is, it's not obvious to me at first the break doesn't get executed.  Also, when leaving such code in place, best to leave a TODO explaining why.

Reply:
Sounds good, I will make the change